components/containerTableRow/index.jsx

```diff
@@ -64,13 +65,46 @@ export function handleAction(resourceIri, containerIri, router) {
   };
 }
 
-export default function ContainerTableRow({ resource }) {
+export function handlePreselection(resourceIri, containerIri, router) {
+  const action = DETAILS_CONTEXT_ACTIONS.DETAILS;
+  return (async () => {
+    await resourceContextRedirect(action, resourceIri, containerIri, router);
+  })();
+}
+
+export default function ContainerTableRow({
+  resource,
+  container,
+  preselected,
+}) {
   const classes = useStyles();
   const bem = useBem(classes);
   const { name, iri } = resource;
   const router = useRouter();
   const { currentUri } = useContext(PodLocationContext);
-  const isActive = router.query.resourceIri === iri;
+  const isActive = router.query.resourceIri === iri || preselected;
```

Comment:
Is a `preselected` parameter necessary? It seems as though you could just do a router redirect with the resource iri in the querystring.

Reply:
Yes, I think this approach is better. Also, might be worth doing a "lookup" of the resource, i.e. do a `getResourceInfo` and check if the resource is a container or not from the response (by checking the URL that is returned - should verify whether it's a container or a non-container).

Reply:
@ajacksified as per discussion I have removed `preselected` from the `isActive` condition.

@megoth I'm checking below whether it is a container using solid-client's `isContainer` in order to decide whether to redirect to the details drawer or not. Is this what you mean?

Reply:
`isContainer` on the iri directly works in most situations, but I encountered an edge-case when I worked on this in the PodNavigator where users could type the URL to a folder _without_ the slash, and if we don't double-check by doing a lookup (i.e. fetch the resource using `useResourceInfo`) and use `isContainer` on the returned URL (which will always be "correct" in terms of having an ending slash or not).

Reply:
In this case we're getting the iri from the bookmark, which gets it from the container table row when the bookmark is created, which gets it from the dataset. Wouldn't it be overkill to fetch the resource info since it does not involve the user manually typing the iri?

Reply:
Ah, yes, for bookmarks it probably doesn't make sense 👍